woocommerce/woocommerce-ios
# Order Details: handle fully refunded order status

Closes #1750.

In this PR the Order Details page now displays the correct information when an order has been refunded and has the `Refunded` status. 

Design reference:
<img src="https://user-images.githubusercontent.com/1062444/72755999-8ff95400-3b91-11ea-8c47-2f8b41098bed.png" width="400" />

### Changes
1. `AggregateOrderItem`s are now filtered to remove any objects where the quantity is zero. That means the Product & Qty section no longer display table cells like this:

<img src="https://user-images.githubusercontent.com/1945542/72784592-1f6f3780-3c64-11ea-8c20-fce57d795258.png" width="350" />

2. There is an edge case where a user can edit the order status and set it to `Refunded`. That order status has been filtered from the edit statuses view controller.

<img src="https://user-images.githubusercontent.com/1062444/72841594-50e40500-3c5c-11ea-9fd4-723c7c87d950.png" width="350" />

If a user did this, it would send the new status to the API. The API would then treat the order as though a "manual refund by amount" was requested. Manual refunds happening through this flow do not report order items as refunded. That means the UI would draw in without a list of order items in the Product & Qty section, and without a Refunded Items section. It looks like this:

<img src="https://user-images.githubusercontent.com/1062444/72841512-27c37480-3c5c-11ea-957e-42a209eed280.png" width="350" />

### To test
The goal is to create a "refund by item" and check the fully refunded order details page.

1. On your test site, have an order with multiple product types. Make sure to note the Order Number.
2. Find the order and "refund by item".  Refund all items, all shipping charges, and all taxes, to trigger the `Refunded` status to update on the website. (Or if you want to watch the Order Details page change, make multiple refunds to see the "partially refunded" state)
5. Check out the develop branch and navigate in the app to: Orders tab > Order Number (Order Details screen). Note that the order item cells with zero quantities appear in the Product & Qty section.
6. Check out this branch. Navigate to the same location. The order items with zero quantities should not appear. The Refunded Products section should have the correct count of order items.

Update release notes:
